packages/components/inputs/radio-input/src/radio-option.styles.ts

```diff
@@ -133,25 +120,96 @@ const getLabelCursor = (props: TStylesProps) => {
   return 'pointer';
 };
 
-const getLabelStyles = (props: TStylesProps) => css`
-  align-items: center;
-  color: ${getLabelColor(props)};
-  cursor: ${getLabelCursor(props)};
+const RadioOptionKnob = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOptionWhenChecked};
+  height: ${designTokens.heightForRadioInputOptionWhenChecked};
+  border-radius: 50%;
+  background-color: ${(props) => getKnobColor(props)};
+`;
+
+const RadioOptionBorder = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOption};
+  height: ${designTokens.heightForRadioInputOption};
+  border-radius: 50%;
+  background-color: ${designTokens.backgroundColorForInput};
+  border: ${designTokens.borderForRadioInputOption} solid
+    ${(props) => getBorderColor(props)};
   display: flex;
-  position: relative;
-  &:hover svg *[data-style='radio-option__border'] {
-    stroke: ${getSvgLabelBorderStroke(props)};
+  align-items: center;
+  justify-content: center;
+`;
+
+const Input = styled.input`
+  &:focus + div > ${RadioOptionBorder} {
+    border-color: ${designTokens.borderColorForInputWhenFocused};
+  }
+`;
+
+const getNewThemeHoverStyles = (props: TStylesProps) => {
+  const hoverStyles = css`
+    [class*='getContainerStyles'] {
```

Comment:
Any suggestions for an alternative solution so would be most welcome 🙏

Reply:
What are you trying to do here? 🤔

Reply:
All in all it turned out that using this class attribute selector makes no sense. Emotion production class names are hashed differently than in development mode. And that is why the hover effect hasn't been working in the preview. Thanks for catching the bug.

Reply:
In the CheckboxInput I didn't need to do this...maybe worth using the same/similar implementation for the elements structure and styles? 🤔

Reply:
I noticed that checkbox and radio inputs work a bit differently. Checkbox doesn't have hover effect when hovering over the label (and radio input does). Not sure if needs to be unified and if yes then what way 🤔 And if for both the old and the new theme.

Reply:
Ah, good point I didn't think about it.

@FilPob Should the hover effect be triggered also when hovering over the label?

Reply:
Oh missed this one. I think it actually looks great when triggering the hover also when hovering just the label. Would be nice to add that also in the Checkbox Input @emmenko

Reply:
Sure, here you go: https://github.com/commercetools/ui-kit/pull/2427

Reply:
@kark Up to you to see if we can use a similar implementation structure.

Reply:
Thanks, I used the same targeting. Updated here 7cff85436acf8dd54d2869823cae78ccb1c64663